TEI.js

```diff
@@ -182,7 +182,7 @@ function generateItem(item, teiDoc) {
 
 	if (Zotero.getOption("Generate XML IDs")) {
 		if (!generatedItems[item.uri]) {
-			var xmlid = genXMLId(item);
+			var xmlid = item.citekey || genXMLId(item); // use Better BibTeX for Zotero citation key if available
```

Comment:
@dstillman what's your view on this?

Reply:
I fully understand Zotero wouldn't want to do stuff that specifically targets BBT. Are there prospective thoughts on the name for the citation key field when it comes to Zotero natively? I wouldn't mind renaming my field to that ahead of time.

Reply:
I should perhaps explain that I'm proposing this because there is a group of us using the TEI translator for a few projects, and we need something more reliable and configurable than the `xml:id` generation in the TEI translator. This is the best approach to improving it that I can think of without breaking existing workflows.